### PR TITLE
Prevent "array to string conversion" errors when migrating basic entities

### PR DESCRIPTION
**Pull request by leofeyer**
We do not have unit tests for migrations, so no, I cannot add a unit test to prevent regressions. 😄

### CODE REVIEW

**Comment by ausi** [comment]
Where did the `array to string conversion` happen here before? Do you have a column value that I can use to reproduce the issue?

**Comment by leofeyer** [comment]
This is what the array looks like:

```
array:36 [
  0 => array:5 [
    0 => "Time"
    1 => "Room"
    2 => "Session/Workshop"
    3 => "Video"
    4 => "Download/Link"
  ]
  1 => array:5 [
    0 => "Thursday<br>10:00 am - 11:30 am"
    1 => "Main Stage"
    2 => "<strong>Begrüßung & Keynote</strong><br>Leo Feyer"
    3 => "<a href="https://www.youtube.com/watch?v=tKL5nQxtGEY" title="Begrüßung [&] Keynote by Leo Feyer" target="_blank">Video on Youtube / German</a>"
    4 => "<a href="https://leofeyer.com/files/slides/2015/konferenz/#1" title="Keynote by Leo Feyer" target="_blank">Slides / German</a>"
  ]
  2 => array:5 [
    0 => "Thursday<br>11:45 am - 12:45 pm"
    1 => "Main Stage"
    2 => "<strong>Warum eigentlich ständig dieser Composer?</strong><br>Nicky Hoff"
    3 => "<a href="https://www.youtube.com/watch?v=VkeoIuCjGZ0" title="Warum eigentlich ständig dieser Composer? by Nicky Hoff" target="_blank">Video on Youtube / German</a>"
    4 => "-"
  ]
  …
```

It seems to be the time table of an old Contao conference.

**Comment by leofeyer** [comment]
But independent from the value, note that we do already handle the array case:

https://github.com/contao/contao/blob/96441e3b97579023fc10c1eb13571b8778076a4f/core-bundle/src/Migration/Version500/AbstractBasicEntitiesMigration.php#L84-L88

We just must not apply `StringUtil::restoreBasicEntities()` if `$value` is an array.

File: core-bundle/src/Migration/Version500/AbstractBasicEntitiesMigration.php [hunk 1]
```diff
@@ -81,10 +81,12 @@ public function run(): MigrationResult
             );
 
             foreach ($values as $id => $value) {
-                $value = StringUtil::restoreBasicEntities(StringUtil::deserialize($value));
+                $value = StringUtil::deserialize($value);
 
                 if (\is_array($value)) {
-                    $value = serialize($value);
+                    $value = serialize(array_map(static fn ($v) => StringUtil::restoreBasicEntities($v), $value));
```
Comment (ausi) [inline core-bundle/src/Migration/Version500/AbstractBasicEntitiesMigration.php]: This needs to be done recursive in order to work with even deeper arrays I think.

```suggestion
                    array_walk_recursive($value, static fn (&$v) => $v = StringUtil::restoreBasicEntities($v));
                    $value = serialize($value);
```

**Review by ausi — APPROVED** [review]

**Pull request merged**